src/yoti_common/anchor.processor.js

```diff
@@ -0,0 +1,176 @@
+'use strict';
+
+const forge = require('node-forge');
+const protoRoot = require('../proto-root');
+
+const YotiAnchor = function main(anchorObj) {
+  this.value = anchorObj.value;
+  this.subType = anchorObj.subType;
+  this.signedTimeStamp = anchorObj.signedTimeStamp;
+  this.originServerCerts = anchorObj.originServerCerts;
+};
+
+YotiAnchor.prototype = {
+  getValue() { return this.value; },
+  getSubType() { return this.subType; },
+  getSignedTimeStamp() { return this.signedTimeStamp; },
+  getOriginServerCerts() { return this.originServerCerts; },
+};
+
+const YotiSignedTimeStamp = function main (timeStampObj) {
+  this.version = timeStampObj.version;
+  this.timestamp = timeStampObj.timestamp;
+};
+
+YotiSignedTimeStamp.prototype = {
+  getVersion() { return this.version; },
+  getTimestamp() { return this.timestamp; }
+};
+
+module.exports.AnchorProcessor = class AnchorProcessor {
+  /**
+   * Extract matching Attribute Anchors from list.
+   *
+   * @param anchors
+   *
+   * @returns {Array}
+   */
+  static process(anchors) {
+    // This will contain a list of anchor values as objects
+    const anchorsData = {
+      sources: [],
+      verifiers: [],
+      processedValues: { // This is used to make sure the anchors values are unique
+        sources: [],
+        verifiers: [],
+      },
+    };
+
+    let anchorObj = {
+      value: '',
+      subType: '',
+      signedTimeStamp: '',
+      originServerCerts: '',
+    };
+
+    for (let i = 0; i < anchors.length; i += 1) {
+      const protoAnchor = anchors[i];
+      const certificatesList = protoAnchor.originServerCerts;
+
+      anchorObj = Object.assign(anchorObj, protoAnchor);
+      anchorObj.signedTimeStamp = this.processSignedTimeStamp(protoAnchor.getSignedTimeStamp());
+
+      for (let n = 0; n < certificatesList.length; n += 1) {
+        const certArrayBuffer = certificatesList[n];
```

Comment:
The content of this for can also extracted into a function that either returns the value of the anchor or null.
The caller should loop over the certificates and set the value in the anchorObj only if it is not null.